pkg/ocm/billing.go

```diff
@@ -34,22 +36,63 @@ func (c *Client) GetBillingAccounts() ([]string, error) {
 		return nil, handleErr(quotaCostResponse.Error(), err)
 	}
 
-	var billingAccounts []string
+	var billingAccounts []*v1.CloudAccount
 	for _, item := range quotaCostResponse.Items().Slice() {
-		for _, account := range item.CloudAccounts() {
-			if account.CloudProviderID() == "aws" && !helper.Contains(billingAccounts, account.CloudAccountID()) {
-				billingAccounts = append(billingAccounts, account.CloudAccountID())
-			}
-		}
+		billingAccounts = append(billingAccounts, item.CloudAccounts()...)
 	}
 
 	if len(billingAccounts) == 0 {
-		return billingAccounts, errors.New("no billing accounts found")
+		return billingAccounts, errors.New("No billing account associated. " +
+			"Go to https://docs.openshift.com/rosa/rosa_architecture/rosa-understanding.html" +
+			" to learn how to associate your billing account." +
+			" You must associate a billing account to continue.")
 	}
 
 	return billingAccounts, nil
 }
 
+func GenerateBillingAccountsList(cloudAccounts []*v1.CloudAccount) []string {
+	var billingAccounts []string
+	for _, account := range cloudAccounts {
+		if account.CloudProviderID() == "aws" && !helper.ContainsPrefix(billingAccounts, account.CloudAccountID()) {
+			var contractString string
+			if HasValidContracts(account) {
+				contractString = " [Contract enabled]"
+			}
+			contractEnabledBillingAccount := account.CloudAccountID() + contractString
+			billingAccounts = append(billingAccounts, contractEnabledBillingAccount)
+		}
+	}
+	return billingAccounts
+
+}
+
+func GetNumsOfVCPUsAndClusters(dimensions []*v1.ContractDimension) (int, int) {
```

Comment:
Can we get a unit test for this please.

Reply:
Yep we do have existing unit test: https://github.com/openshift/rosa/pull/1540/files#diff-d3298d977cc65d850ac089fe355ea4c5ab4269c239bf9e0203a04be16b80ea6cR177